source/MBF.SAMD51.SystemCore.pas

```diff
@@ -25,6 +25,35 @@ interface
 {$INCLUDE MBF.ARM.SystemCore.inc}
 {$UNDEF INTERFACE}
 
+{$ifdef freertos_fat}
+
+  {$LINKLIB libfreertos_wio.a}
+  {$LINKLIB libstdc++_nano.a,static}
+  {$LINKLIB libm.a,static}
+  {$LINKLIB libc_nano.a,static}
+  {$LINKLIB libgcc.a,static}
+  {$LINKLIB libnosys.a,static}
+
+  procedure BasicSystemInit; cdecl; external name 'SAMD51SystemInit';
+
+  procedure BasicTickHandler; cdecl; external name 'SysTick_DefaultHandler';
+  function  FreeRTOSSysTickHook:integer; cdecl; external name 'sysTickHook';
+
+  function  roundf(x:single):Cardinal; cdecl; external name 'lroundf';
```

Comment:
Have you checked if now the math functions work correctly? When I understood Jeppe correctly we should have sin cos & friends from math unit, no need to borrow them from C-Code. I thin k I checked but honestly do not remember the result, but I think it worked....

Reply:
Math does not work without libm unfortunately.